From ee0061bca474cfbbb0a3ec06ea0420f145e36050 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>
Date: Tue, 21 Jan 2020 16:13:49 -0600
Subject: [PATCH 26/37] MLK-23258-3 dts: Fix PCIE suspend/resume issue

commit e1eb607b6bc0724ae0ec5ae1e166091c8049ce99 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix the parent-child power domain dependency to handle different
PCIE usecases.

Signed-off-by: Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>
(cherry picked from commit 29fac883f2f0fc5d8d5769dfc105b9d94c4863a1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi | 25 ++++++++++---------
 include/dt-bindings/soc/imx8_pd.h             |  2 ++
 2 files changed, 15 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index d51dc073b5c6..cc30fd515263 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -490,25 +490,26 @@
 						#address-cells = <1>;
 						#size-cells = <0>;
 
-						pd_pcie1: PD_HSIO_PCIE_B {
-							reg = <SC_R_PCIE_B>;
-							#power-domain-cells = <0>;
-							power-domains =<&pd_pcie0>;
-							#address-cells = <1>;
-							#size-cells = <0>;
+					};
 
-							pd_serdes1: PD_HSIO_SERDES_1 {
-								reg = <SC_R_SERDES_1>;
+					pd_serdes1: PD_HSIO_SERDES_1 {
+						reg = <SC_R_SERDES_1>;
+						#power-domain-cells = <0>;
+						power-domains =<&pd_serdes0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						pd_pcie1: PD_HSIO_PCIE_B {
+								reg = <SC_R_PCIE_B>;
 								#power-domain-cells = <0>;
-								power-domains =<&pd_pcie1>;
+								power-domains =<&pd_serdes1>;
 								#address-cells = <1>;
 								#size-cells = <0>;
 
-								pd_sata0: PD_HSIO_SATA_0 {
+							pd_sata0: PD_HSIO_SATA_0 {
 									reg = <SC_R_SATA_0>;
 									#power-domain-cells = <0>;
-									power-domains =<&pd_serdes1>;
-								};
+									power-domains =<&pd_pcie1>;
 							};
 						};
 					};
diff --git a/include/dt-bindings/soc/imx8_pd.h b/include/dt-bindings/soc/imx8_pd.h
index d955bcf94914..f5348fd80ad1 100644
--- a/include/dt-bindings/soc/imx8_pd.h
+++ b/include/dt-bindings/soc/imx8_pd.h
@@ -72,6 +72,8 @@
 #define PD_HSIO_PCIE_B              hsio_pcie1
 #define PD_HSIO_SATA_0              hsio_sata0
 #define PD_HSIO_GPIO                hsio_gpio
+#define PD_HSIO_SERDES_0            hsio_serdes_0
+#define PD_HSIO_SERDES_1            hsio_serdes_1
 
 #define PD_LCD_0                    lcd0_power_domain
 #define PD_LCD_0_I2C_0              lcd0_i2c0
-- 
2.17.1

